src/build/env.py

```diff
@@ -240,18 +264,22 @@ def _create_isolated_env_venv(path: str) -> tuple[str, str]:
         with warnings.catch_warnings():
             if sys.version_info[:3] == (3, 11, 0):
                 warnings.filterwarnings('ignore', 'check_home argument is deprecated and ignored.', DeprecationWarning)
-            venv.EnvBuilder(with_pip=True, symlinks=symlinks).create(path)
+            venv.EnvBuilder(with_pip=not _valid_global_pip(), symlinks=symlinks).create(path)
     except subprocess.CalledProcessError as exc:
         raise FailedProcessError(exc, 'Failed to create venv. Maybe try installing virtualenv.') from None
 
     executable, script_dir, purelib = _find_executable_and_scripts(path)
 
     # Get the version of pip in the environment
-    if not _has_valid_pip(purelib):
+    if not _valid_global_pip() and not _has_valid_pip(path=[purelib]):
         _subprocess([executable, '-m', 'pip', 'install', f'pip>={_minimum_pip_version()}'])
 
     # Avoid the setuptools from ensurepip to break the isolation
-    _subprocess([executable, '-m', 'pip', 'uninstall', 'setuptools', '-y'])
+    if _valid_global_pip():
```

Comment:
Redundant, setuptools is not installed if `with_pip` is false. (It is also not installed on Python 3.12+.)

Reply:
Ah, okay, yes. Removed this branch.